README.md

```diff
@@ -1,117 +1,200 @@
 <!-- markdownlint-disable MD004 -->
-# Overview
+
+# AKS middleware for Go
+
+<!-- vscode-markdown-toc -->
+* 1. [Usage](#Usage)
+* 2. [gRPC server](#gRPCserver)
+ 	* 2.1. [requestid](#requestid)
+ 	* 2.2. [ctxlogger (applogger)](#ctxloggerapplogger)
+ 	* 2.3. [autologger (api request/response logger)](#autologgerapirequestresponselogger)
+ 	* 2.4. [recovery](#recovery)
+ 	* 2.5. [protovalidate](#protovalidate)
+* 3. [gRPC client](#gRPCclient)
+ 	* 3.1. [mdforward](#mdforward)
+ 	* 3.2. [autologger (api request/response logger)](#autologgerapirequestresponselogger-1)
+ 	* 3.3. [retry](#retry)
+* 4. [HTTP server](#HTTPserver)
+ 	* 4.1. [requestid](#requestid-1)
+ 	* 4.2. [ctxlogger (applogger)](#ctxloggerapplogger-1)
+ 	* 4.3. [logging (api request/response logger)](#loggingapirequestresponselogger)
+ 	* 4.4. [recovery](#recovery-1)
+ 	* 4.5. [inputvalidate](#inputvalidate)
+* 5. [HTTP client via Azure SDK](#HTTPclientviaAzureSDK)
+ 	* 5.1. [mdforward](#mdforward-1)
+ 	* 5.2. [policy (api request/response logger)](#policyapirequestresponselogger)
+ 	* 5.3. [retry](#retry-1)
+* 6. [HTTP client via Direct HTTP request](#HTTPclientviaDirectHTTPrequest)
+ 	* 6.1. [mdforward](#mdforward-1)
+ 	* 6.2. [Restlogger (api request/response logger)](#Restloggerapirequestresponselogger)
+ 	* 6.3. [retry](#retry-1)
+* 7. [Project](#Project)
+ 	* 7.1. [Contributing](#Contributing)
+ 	* 7.2. [Trademarks](#Trademarks)
+
+<!-- vscode-markdown-toc-config
+	numbering=true
+	autoSave=true
+	/vscode-markdown-toc-config -->
+<!-- /vscode-markdown-toc -->
 
 This directory is the root of the aks-middleware module. It implements interceptors to aid common service tasks. See the list below for details.
 
-# Usage
+## 1. <a name='Usage'></a>Usage
 
-After the code is generated, run the following command in the root directory of this module.
+Run the following command in the root directory of this module.
 
 ```bash
 make
 ```
 
-# requestid
+## 2. <a name='gRPCserver'></a>gRPC server
+
+### 2.1. <a name='requestid'></a>requestid
 
 ServerInterceptor. It adds x-request-id to MD if there is no such entry. This interceptor needs to be registered first so that its request-id can be used by autologger and applogger.
 
-# mdforward
+### 2.2. <a name='ctxloggerapplogger'></a>ctxlogger (applogger)
 
-ClientInterceptor. It propagates MD from incoming to outgoing. Only need to be used in servers. No need to be used in a pure client app.
+ServerInterceptor. It adds a logger to ctx with important information (e.g., request-id, method-name) already populated. App's handler code can get the logger and output additional information. Needs to be registered after the requestid interceptor.
```

Comment:
Since we are putting this under the gRPC server function, mentioning server interceptor here feels redundant

Reply:
Done.

Reply:
Done